java/src/org/openqa/selenium/grid/sessionqueue/remote/RemoteNewSessionQueue.java

```diff
@@ -171,7 +171,12 @@ public void complete(
     }
 
     HttpTracing.inject(tracer, tracer.getCurrentContext(), upstream);
-    client.with(addSecret).execute(upstream);
+    HttpResponse response = client.with(addSecret).execute(upstream);
+    if (response.isSuccessful()) {
+      return Values.get(response, Boolean.class);
+    } else {
+      return false;
```

Comment:
`return Values.get(response, Boolean.class);` should be enough here, there is an internal error handling in `Values.get`

Reply:
Ok, was not sure about it, I will change it